server/src/main/java/org/opensearch/plugins/Plugin.java

```diff
@@ -269,4 +269,13 @@ public void close() throws IOException {
     public Collection<IndexSettingProvider> getAdditionalIndexSettingProviders() {
         return Collections.emptyList();
     }
+
+    /**
+     * Returns the {@link SecureSettingsFactory} instance that could be used to configure the
+     * security related components (fe. transports)
+     * @return the {@link SecureSettingsFactory} instance
+     */
+    public Optional<SecureSettingsFactory> getSecureSettingFactory(Settings settings) {
```

Comment:
@reta Would it be possible to create a companion PR in the security plugin to show how this extension point can be used to provide the necessary settings?

By having this be in the Plugin.java class does this mean that any plugin can potentially provide a settings factory and not only NetworkPlugins?

Reply:
> @reta Would it be possible to create a companion PR in the security plugin to show how this extension point can be used to provide the necessary settings?

Absolutely

> By having this be in the Plugin.java class does this mean that any plugin can potentially provide a settings factory and not only NetworkPlugins?

So the `SecureSettingsFactory` is not tight only to transports, hence it is not specific to network protocols. Also, the idea is that some plugins may only provide some security related settings but not necessarily all of them